pytket/pytket/__init__.py

```diff
@@ -25,4 +24,7 @@
 import pytket.transform
 import pytket.telemetry
 
+from pytket._version import __version__
+
 __path__ = __import__("pkgutil").extend_path(__path__, __name__)  # type: ignore
+__version__ = ".".join(__version__.split(".")[:3])
```

Comment:
Is this truncation needed? For released versions it will make no difference, and for dev versions is there any reason not to have the full version string?

Reply:
No particular reasons really. I can make a case for different versions if I can discriminate between the two ?

Reply:
Thing is I haven't found an easy way to format the version string with `setuptools_scm` in the `pyproject.toml`_i.e._ to extract the base, commit id etc. which is made easy with `dunamai` for instance. However, there is no easy `dunamai` integration with `pyproject.toml` unfortunately.

Reply:
I'm just suggesting we leave it as it is, e.g. "0.17.1.dev12+g6129ff39" for a dev version. Is there any problem with that? (For a release version it would just be "0.17.1" or whatever.)

Reply:
(i.e. don't add line 30)

Reply:
Oh OK well sure. I thought release tags where under the same kind of long format. But it they are under the "X.Y.Z" form only then it makes sense.